### DISCUSSION TITLE
Partial `serialize`

### DISCUSSION THREAD

**Discussion by rikbrown** [47ng/nuqs#639] [Q&A] [answered]
Wondering if there's a good pattern for generating the URL given a partial input. For example, if I know I want to change the `page` parameter to `3`, but I want the *full* URL based on the current URL e.g. `x/y/z?foo=bar` becoming `x/y/z/?foo=bar&page=3`.

Asking because I'm thinking about situations I want to dynamically `router.prefetch` (or honestly just use a `Link` component) but in the context I don't have the full set of query parameters, just that I want to change the page.

**Comment by franky47** [comment, marked as answer]
Serialize can takes a base as its first argument, you can feed it:
- A string (containing either a pathname, serialized search params on their own, or a combination of both)
- A URLSearchParams object (could come from `useSearchParams` for the current ones)
- a URL object

The second argument is the map of search params you want to serialize/merge on top of the base.
